pkg/vpcmodel/commonConnectivity.go

```diff
@@ -17,15 +124,38 @@ func (connectivityMap GeneralConnectivityMap) updateAllowedConnsMap(src, dst VPC
 	connectivityMap[src][dst] = conn
 }
 
-func (connectivityMap GeneralConnectivityMap) updateMap(connectivityMap2 GeneralConnectivityMap) {
+func (statefulConnMap GeneralStatefulConnectivityMap) updateMap(connectivityMap2 GeneralStatefulConnectivityMap) {
 	for src, nodeConns := range connectivityMap2 {
 		for dst, conns := range nodeConns {
-			connectivityMap.updateAllowedConnsMap(src, dst, conns)
+			statefulConnMap.updateAllowedStatefulConnsMap(src, dst, conns)
 		}
 	}
 }
-func (connectivityMap GeneralConnectivityMap) copy() GeneralConnectivityMap {
-	newConnectivityMap := GeneralConnectivityMap{}
-	newConnectivityMap.updateMap(connectivityMap)
+func (statefulConnMap GeneralStatefulConnectivityMap) copy() GeneralStatefulConnectivityMap {
+	newConnectivityMap := GeneralStatefulConnectivityMap{}
+	newConnectivityMap.updateMap(statefulConnMap)
 	return newConnectivityMap
 }
+
+// it is assumed that the components of extendedConn are legal connection.Set, namely not nil
+func (statefulConnMap GeneralStatefulConnectivityMap) updateAllowedStatefulConnsMap(src, dst VPCResourceIntf, extendedConn *ExtendedSet) {
+	if _, ok := statefulConnMap[src]; !ok {
+		statefulConnMap[src] = map[VPCResourceIntf]*ExtendedSet{}
+	}
+	statefulConnMap[src][dst] = extendedConn
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////
+
+// todo: following functionality needs to be moved to package connection with member instead of parms passing
+
+func newTCPSet() *connection.Set {
+	return connection.TCPorUDPConnection(netp.ProtocolStringTCP, connection.MinPort, connection.MaxPort,
+		connection.MinPort, connection.MaxPort)
+}
+
+func partitionTCPNonTCP(conn *connection.Set) (tcp, nonTCP *connection.Set) {
+	tcpFractionOfConn := newTCPSet().Intersect(conn)
+	nonTCPFractionOfConn := conn.Subtract(tcpFractionOfConn)
+	return tcpFractionOfConn, nonTCPFractionOfConn
+}
```

Comment:
partitionTCPNonTCP() should be called in another constructor

Reply:
Not sure I follow; construction to what? the returned objects here are Set. Anyways `partitionTCPNonTCP()` will not stay here after the next PR; it will be moved to models.

Reply:
Added `NewConnWithStatefulGivenTCPStatefulAndNonTCP`